crates/egui/src/text_selection/text_cursor_state.rs

```diff
@@ -291,6 +292,18 @@ pub fn is_word_char(c: char) -> bool {
     c.is_ascii_alphanumeric() || c == '_'
 }
 
+/// Determines whether a character is a component of a string.
+///
+/// Punctuation marks that are commonly used with strings are considered string components.
+pub fn is_string_component(c: char) -> bool {
```

Comment:
This name is confusing. Any `char` can be part of a `String`, so it's not clear what "string" means in this context.

Reply:
Since I don't speak English, I don't know what would be better.

Reply:
What is it trying to accomplish? Should this be a change to `is_word_char` instead?

Reply:
A new name was used so as not to affect the existing code.
but, Changed to `is_word_char()`.

Reply:
> What is it trying to accomplish? Should this be a change to `is_word_char` instead?

A new name was used so as not to affect the existing code.
but, Changed to is_word_char().